cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -83,7 +83,8 @@
   private static final Type MAP_STRING_STRING_TYPE = new TypeToken<Map<String, String>>() { }.getType();
   private static final String TYPE_APP_META = "appMeta";
   private static final String TYPE_STREAM = "stream";
-  private static final String TYPE_RUN_RECORD_STARTED = "runRecordStarted";
+  private static final String TYPE_RUN_RECORD_STARTED = "runRecordStarting";
```

Comment:
Agreed, it is confusing. The reason I switched the naming around was because STARTED and STARTING seemed too similar and might be more confusing.

Especially because STARTED means `recordProgramRunning` , i.e `ProgramRunStatus.RUNNING`.

However, we do use somewhat similar terminology elsewhere, so perhaps it would be better to adopt this here as well.